include/alp/type_traits.hpp

```diff
@@ -148,6 +185,71 @@ namespace alp {
 
 	} // end namespace alp::internal
 
+	/**
+	 * Used to get a structure type of the given ALP container
+	 *
+	 * @tparam T The ALP container to inspect.
+	 *
+	 */
+	template< typename Container >
+	struct inspect_structure {};
+
+	/**
+	 * Used to get a View type of the given ALP container
+	 *
+	 * @tparam T The ALP container to inspect.
+	 *
+	 */
+	template< typename Container >
+	struct inspect_view {};
+
+
+	namespace internal {
+
+		template< typename T >
+		struct is_view_over_concrete_container : is_view_over_concrete_container< typename inspect_view< T >::type::applied_to > {
```

Comment:
What is the difference between this and `is_original`? I think the latter should be enough or am I missing something?